src/tr2/game/creature.c

```diff
@@ -952,3 +956,78 @@ bool Creature_IsAlly(const ITEM *const item)
 {
     return Object_IsObjectType(item->object_id, g_AllyObjects);
 }
+
+int32_t __cdecl Creature_ShootAtLara(
+    ITEM *const item, AI_INFO *const info, BITE *const gun,
+    const int16_t extra_rotation, const int32_t damage)
+{
+    const CREATURE *const creature = item->data;
+    ITEM *const target_item = creature->enemy;
+
+    bool is_targetable;
+    bool is_hit;
+    if (info->distance > CREATURE_SHOOT_RANGE
+        || !Creature_CanTargetEnemy(item, info)) {
+        is_targetable = false;
+        is_hit = false;
+    } else {
+        int32_t distance =
+            (((target_item->speed * Math_Sin(info->enemy_facing)) >> W2V_SHIFT)
+             * CREATURE_SHOOT_RANGE)
+            / CREATURE_SHOOT_TARGETING_SPEED;
+        distance = info->distance + SQUARE(distance);
+        if (distance > CREATURE_SHOOT_RANGE) {
+            is_hit = false;
+        } else {
+            const int32_t chance = CREATURE_SHOOT_HIT_CHANCE
+                + (CREATURE_SHOOT_RANGE - info->distance)
+                    / (CREATURE_SHOOT_RANGE / 0x5000);
+            is_hit = Random_GetControl() < chance;
+        }
+        is_targetable = true;
+    }
+
+    int16_t fx_num = NO_ITEM;
+    if (target_item == g_LaraItem) {
+        if (is_hit) {
+            fx_num = Creature_Effect(item, gun, Effect_GunHit);
+            Item_TakeDamage(target_item, damage, true);
+        } else if (is_targetable) {
+            fx_num = Creature_Effect(item, gun, Effect_GunMiss);
+        }
+    } else {
+        fx_num = Creature_Effect(item, gun, Effect_GunShot);
+        if (is_hit) {
+            Item_TakeDamage(target_item, damage / 10, true);
+        }
+    }
+
+    if (fx_num != NO_ITEM) {
+        g_Effects[fx_num].rot.y += extra_rotation;
+    }
+
+    GAME_VECTOR start = {
+        .pos = {
+            .x = item->pos.x,
+            .y = item->pos.y - STEP_L * 3,
+            .z = item->pos.z,
+        },
+        .room_num = item->room_num,
+    };
+
+    GAME_VECTOR target = {
+        .pos = {
+            .x = item->pos.x,
```

Comment:
Should this perhaps be `target_item` - could explain the window crash?

Reply:
@lahm86 thanks, that's a good catch, and you're absolutely right.